include/matx/transforms/chol/chol_cuda.h

```diff
@@ -161,9 +161,15 @@ class matxDnCholCUDAPlan_t : matxDnCUDASolver_t {
           this->d_info + i);
 
       MATX_ASSERT(ret == CUSOLVER_STATUS_SUCCESS, matxSolverError);
-      
-      // This will block. Figure this out later
-      cudaMemcpy(&info, this->d_info + i, sizeof(info), cudaMemcpyDeviceToHost);
+    }
+
+    std::vector<int> h_info(this->batch_a_ptrs.size());
+    cudaMemcpyAsync(h_info.data(), this->d_info, sizeof(int) * this->batch_a_ptrs.size(), cudaMemcpyDeviceToHost, stream);
+    
+    // This will block. Figure this out later
+    cudaStreamSynchronize(stream);
+
+    for (const auto& info : h_info) {
       MATX_ASSERT(info == 0, matxSolverError);
```

Comment:
While you're here it might be good to add a more descriptive string onto this. The docs say: 
```
l. If info = -i, the i-th parameter is wrong (not counting handle). If info = i, the leading minor of order i is not positive definite.
```

Reply:
Sure, I'll update these and the rest (also the LAPACK error checks)